### PR TITLE
feat: add s3 raw files path and darty path facilities of EventModels

### PR DESCRIPTION
**Pull request by slezier**
in order to be used in pubstack-cli.

Looks artificial, i'm not sure to make the good PR.

### CODE REVIEW

**Comment by slezier** [comment]
@erwanncloarec let's judge

File: rust/src/event.rs [hunk 1]
```diff
@@ -41,6 +42,53 @@ impl FromStr for EventModel {
...
+    pub fn get_parquet_path_of_model(&self) -> String {
+        let model = self.to_string();
+        format!("parquet/{model}/")
+    }
+
+    #[inline(always)]
+    pub fn generate_parquet_path(&self, scope: &str, date: &str) -> String {
+        let parquet_path_of_model = self.get_parquet_path_of_model();
+        format!("{parquet_path_of_model}scope={scope}/date={date}/")
+    }
```
Comment (erwanncloarec) [inline rust/src/event.rs]: I think it's weird to have those path here. We have a lot of paths all different from one component to another, I don't see why the model library should have this. On the technical point of view, you can use a trait that you implement for `EventModel` in the cli. We could also think of a Darty lib to be reused in the cli for ex.

File: rust/src/event.rs [hunk 2]
```diff
@@ -41,6 +42,53 @@ impl FromStr for EventModel {
...
+    #[inline(always)]
+    pub fn to_str(&self) -> &str {
+        match self {
+            EventModel::Auction => "auction",
+            EventModel::Impression => "impression",
+            EventModel::ViewableImpression => "viewability",
+            EventModel::MeasuredImpression => "measured",
+            EventModel::Page => "page",
+        }
+    }
```
Comment (erwanncloarec) [inline rust/src/event.rs:89]: I would have just implemented to_string as it is from a official rust trait, but fine if we need this

File: rust/src/event.rs (hunk 1 above)
Comment (erwanncloarec) [reply rust/src/event.rs]: Also those methods makes the lib import chrono, the fewer imports we have, the better !

Comment (slezier) [reply rust/src/event.rs]: yeap i want to use "augmented" enum  in pubstack cli with all path mapping  stuff. 
It is the only way (i see) to link the EventModel to the treatment.

Comment (slezier) [reply rust/src/event.rs]: I am not comfortable with adding functions mainly relying on one element. It is an hidden object like c way of coding

File: rust/Cargo.toml [hunk 3]
```diff
@@ -14,6 +14,7 @@ serde = {version = "1.0.126", features = ["derive"]}
 tracing = "0.1.29"
 tokio = {version="1.15.0", features=["io-util"]}
 async-compression = { version= "0.3.8", features=["tokio", "gzip"]}
+chrono = {version = "0.4.19", features = []}
```
Comment (MichaelBitard) [inline rust/Cargo.toml:17]: This does **not** deactivate the features :), You have to add something like default-feature = false

**Review by MichaelBitard — APPROVED** [review]
I'm not a fan of the naming, but I don't have any counter proposal though

File: rust/src/event.rs (hunk 1 above)
Comment (slezier) [reply rust/src/event.rs]: @erwanncloarec  change the code ... only iter

File: rust/src/event.rs (hunk 2 above)
Comment (slezier) [reply rust/src/event.rs]: @erwanncloarec  change the code

File: rust/Cargo.toml (hunk 3 above)
Comment (slezier) [reply rust/Cargo.toml]: @MichaelBitard  change  the code in order to host less cli feature (sorry you have to review it again)

**Pull request merged**